### DISCUSSION TITLE
Only run update system when there are changes to an inspected resource

### DISCUSSION THREAD

**Discussion by askor** [jakobhellermann/bevy-inspector-egui#92] [Q&A] [answered]
In the [demo example](https://github.com/jakobhellermann/bevy-inspector-egui/blob/main/examples/demo.rs), a system continuously runs to update entities with the Data resource. This seems inefficient if the project grows, and the number of inspectors and entities becomes large. I would like the update system to only run when I change values in the inspector.

#### How can I only update components when there are changes to the inspected resource?

**Comment by jakobhellermann** [comment, marked as answer]
You can add
```rust
if !data.is_changed() {
    return;
}
```
to your function, `bevy-inspector-egui` should only mark resources as changed if they were actually edited.
